models/meshmodel/registry/registry.go

```diff
@@ -383,6 +388,24 @@ func (rm *RegistryManager) GetModels(db *database.Handler, f types.Filter) ([]v1
 		model.HostID = host.ID
 		model.HostName = host.Hostname
 		model.DisplayHostName = host.Hostname
+
+		if includeComponents {
+			var components []v1alpha1.ComponentDefinitionDB
+			finder := db.Model(&v1alpha1.ComponentDefinitionDB{}).
+				Select("component_definition_dbs.*").
+				Where("component_definition_dbs.model_id = ?", model.ID)
```

Comment:
With this, the component schema would also be included, which is not required in the registry view.

Reply:
Hmm. So, you're proposing to introduce only necessary fields, correct?

Reply:
I think yes, adding a flag can be nice, in case we might want to retrieve other data too.

Reply:
We usually have `trim` param to avoid sending schema if it is not required checkout [doc](https://docs.meshery.io/reference/rest-apis#api-meshmodels-models-model-components)

Reply:
Yes, we do have trim param but for components API not here, since model definitions didn't have schema as such.

@VihasMakwana We want a functionality to selectively retrieve each of the attributes, the fields to retrieve could be passed on as a filter. Though this is something we want but not in this PR. (Something to keep in mind).

Ensuring just the schema is not passed we are good here.

Reply:
Yeah. makes sense

Reply:
I've only requested necessary fields i.e. kind, modelName, apiVersion.
Please re-review @Yashsharma1911 @MUzairS15